Connection.php

```diff
@@ -555,3 +555,26 @@ public function close($code = self::CLOSE_NORMAL, $reason = null)
         return $connection->disconnect();
     }
 }
+
+/**
+ * Register socket wrappers
+ */
+$factory = function($socketUri) {
+    $parsed = parse_url($socketUri);
+
+    $secure = 'wss' === $parsed['scheme'];
+
+    if (isset($parsed['port'])) {
+        $port = $parsed['port'];
+    } else {
+        $port = true === $secure ? 443 : 80;
+    }
+
+    return new WebsocketSocket(
+        'tcp://' . $parsed['host'] . ':' . $port,
```

Comment:
What if `port` index is undefined?
What to do with `user` and `pass` indexes too?